### PR TITLE
ENH: Partial string matching for timestamps with multiindex

### LINKED ISSUE #10331: Partial string matching for timestamps with multiindex
I'm trying to get a slice from a multiindex. I find the behavior pretty inconsistent. Here is a simple dataframe:

```python
test = pd.DataFrame({'l': ['A', 'B', 'C', 'C', 'B', 'A'], 'm': [pd.Timestamp('2014-06-11 14:26:27'), pd.Timestamp('2014-06-11 15:26:27'),  pd.Timestamp('2014-06-11 16:26:27'), pd.Timestamp('2014-06-12 14:26:27'), pd.Timestamp('2014-06-12 15:26:27'),  pd.Timestamp('2014-06-12 16:26:27')]})
```
With a single index, I can select all the data for a given day as follows:
```python
test_single_index = test.set_index(['m'])
#partial string indexing works
test_single_index.loc['2014-06-11']
```
But it doesn't work for a multiindex:
```python
test_multi_index = test.set_index(['m', 'l'])
#would expect this to work
test_multi_index.loc['2014-06-11']
#or this
test_multi_index.loc[('2014-06-11', 'A'),:]

#what works
idx = pd.IndexSlice
test_multi_index.loc[idx['2014-06-11':'2014-06-11',:],:]
test_multi_index.loc[idx['2014-06-11','A'],:]
```
So I can make it work if I specify the slice explicitely, but it would be nice if the behavior for the 1D index carried over to Multiindices.

### PR DESCRIPTION
**Pull request by markroth8**
- [x] closes #10331
- [x] tests added / passed
- [x] passes `git diff upstream/master | flake8 --diff`
- [x] whatsnew entry

Includes implementation, unit tests, documentation fixes and whatsnew.

### CODE REVIEW

File: doc/source/timeseries.rst [hunk 1]
```diff
@@ -422,6 +422,20 @@ We are stopping on the included end-point as it is part of the index
 
       dft.loc['2013-1-15 12:30:00']
 
+DatetimeIndex Partial String Indexing also works on DataFrames with hierarchical indexing (``MultiIndex``). For
+instance:
+
+.. ipython:: python
+
+   dft2 = DataFrame(randn(20,1),
```
Comment (sinhrks) [inline doc/source/timeseries.rst]: Pls use ``pd.`` and ``np.``, see #9886.

File: doc/source/whatsnew/v0.18.1.txt [hunk 2]
```diff
@@ -22,6 +22,8 @@ New features
 Enhancements
 ~~~~~~~~~~~~
 
+Partial string matches on `DateTimeIndex` now work when part of a `MultiIndex`
```
Comment (jreback) [inline doc/source/whatsnew/v0.18.1.txt]: use double-back ticks around ``DatetimeIndex`` (sp as well) (and same around ``MultiIndex``); put the issue reference on the same line

**Comment by jreback** [comment]
looks real nice @markroth8 

just some minor doc-comments. pls rebase on master when you get a chance; master now builds correctly.

**Comment by jreback** [comment]
actually pls also put an example in the this section [here](http://pandas-docs.github.io/pandas-docs-travis/timeseries.html#datetimeindex-partial-string-indexing) as well

**Comment by jreback** [comment]
pls rebase and i'll have a look

File: doc/source/whatsnew/v0.18.1.txt [hunk 3]
```diff
@@ -22,6 +22,8 @@ New features
...
+(:issue:`10331`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.18.1.txt]: pls add a min-example here (you can add a sub-section if you'd like to highlite to setup an example and show how it works)

File: pandas/core/indexing.py [hunk 4]
```diff
@@ -1388,8 +1388,33 @@ def error():
 
         return True
 
+    def _get_partial_string_timestamp_match_key(self, key, labels):
+        """Translate any partial string timestamp matches in key, returning the
+        new key"""
```
Comment (jreback) [inline pandas/core/indexing.py]: add this issue number in a comment here

File: doc/source/whatsnew/v0.18.1.txt (hunk 3 above)
Comment (jreback) [inline doc/source/whatsnew/v0.18.1.txt]: you can pretty much use the example for ``timeseries.rst``

**Comment by jreback** [comment]
@markroth8 can you update/rebase.

File: doc/source/timeseries.rst (hunk 1 above)
Comment (markroth8) [inline doc/source/timeseries.rst]: @sinhrks, Fixed.

File: doc/source/whatsnew/v0.18.1.txt (hunk 2 above)
Comment (markroth8) [inline doc/source/whatsnew/v0.18.1.txt]: Done.

File: doc/source/whatsnew/v0.18.1.txt (hunk 3 above)
Comment (markroth8) [inline doc/source/whatsnew/v0.18.1.txt]: Done.

File: pandas/core/indexing.py (hunk 4 above)
Comment (markroth8) [inline pandas/core/indexing.py]: Done.

**Comment by markroth8** [comment]
@jreback: Incorporated suggestions and rebased / pushed.

**Pull request closed without merging**